### PR TITLE
Add installation to support find_package

### PR DESCRIPTION
**Pull request by pfultz2**
This now can be installed with cmake and can be used, like:

```cmake
find_package(sugar)
```

### CODE REVIEW

**Comment by ruslo** [comment]
For some reason I can't leave the comments in review. Writing them here:

### CMakeLists.txt (line 2)

I don't think it makes any difference in current case but it's a good habit to put `project` right after `cmake_minimum_required`:
* http://cgold.readthedocs.io/en/latest/tutorials/project.html#when-not-declared

### CMakeLists.txt (line 6)

Why do we need `enable_testing`?

### sugar-config.cmake

There is no check of components. If user write `find_package(sugar CONFIG REQUIRED missing-component)` we should report an error. This can be done by `check_required_components`:
* https://github.com/forexample/package-example/blob/cf2ea1d6a209fb9eca2ab83fdd0ac15fe4d3e807/Foo/cmake/Config.cmake.in#L4

**Comment by ruslo** [comment]
@pfultz2 This idea is great, thanks. I was thinking about unification of CMake modules and regular C++ packages (https://docs.hunter.sh/en/latest/overview/anything.html) before, this seems to be a nice solution.

**Comment by pfultz2** [comment]
> Why do we need enable_testing?

It there just in case you want to add testing. I can remove it for now.

**Comment by pfultz2** [comment]
I also have updated this based on your feedback.

File: sugar-config.cmake.in [hunk 1]
```diff
@@ -0,0 +1,4 @@
+@PACKAGE_INIT@
+
+include(cmake/Sugar)
```
Comment (ruslo) [inline sugar-config.cmake.in]: I've tried to use the package but got error:
```
share/sugar/sugar-config.cmake:27 (include):
  include could not find load file:

    cmake/Sugar
```

I guess the solution is to use full path:
```cmake
include("${CMAKE_CURRENT_LIST_DIR}/cmake/Sugar")
```

Comment (pfultz2) [inline sugar-config.cmake.in]: Yea, that's probably better. Updated.

**Comment by ruslo** [comment]
Looks good, merged. Thank you!

**Pull request merged**